src/Symfony/Component/Serializer/Normalizer/AbstractObjectNormalizer.php

```diff
@@ -300,6 +305,7 @@ abstract protected function getAttributeValue(object $object, string $attribute,
      */
     public function supportsDenormalization(mixed $data, string $type, string $format = null /* , array $context = [] */)
     {
+        // @deprecated return true in 7.0
         return class_exists($type) || (interface_exists($type, false) && $this->classDiscriminatorResolver && null !== $this->classDiscriminatorResolver->getMappingForClass($type));
```

Comment:
`null !== $this->classDiscriminatorResolver?->getMappingForClass($type)` for the last part while at it :)

Sure about `return true;` in 7.0?

Reply:
indeed!